[easy] encapsulate fb changes from OSS

kflu opened the pull request:
Summary:
encapsulate fb changes into `torch._inductor.fx_passes.fb`, so that adding new passes (`fb.xxx`) won't need to touch OSS code like so:

```
# in torch/_inductor/fx_passes/pre_grad.py
if config.is_fbcode():
 from .fb import xxx  # every new fb/xxx.py would have needed this change in OSS code base
```

Test Plan: CI

Differential Revision: D51315193



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

kflu commented:
@pytorchbot merge

aakhundov commented on torch/_inductor/fx_passes/pre_grad.py:
```diff
@@ -48,7 +48,7 @@ def lazy_init():
 
     if config.is_fbcode():
-        from .fb import split_cat as split_cat_fb  # type: ignore[import]  # noqa: F401
+        from . import fb  # type: ignore[import]  # noqa: F401
```
Noob q: how does importing this within the function help the outside code? Are there some persistent side effects from importing?

kflu replied:
yes within those python modules (like `split_cat.py`), there's a global function registered with `@register_graph_optimization(...)`. So just importing is enough.

kflu committed:
[easy] encapsulate fb changes from OSS (#113677)

Summary:

encapsulate fb changes into `torch._inductor.fx_passes.fb`, so that adding new passes (`fb.xxx`) won't need to touch OSS code like so:

```
# in torch/_inductor/fx_passes/pre_grad.py
if config.is_fbcode():
 from .fb import xxx  # every new fb/xxx.py would have needed this change in OSS code base
```

Test Plan: CI

Reviewed By: frank-wei, khabinov, chenyang78

Differential Revision: D51315193
